src/java/htsjdk/samtools/sra/SRAAccession.java

```diff
@@ -75,11 +83,34 @@ public static boolean isSupported() {
      * @return true if a string is a valid SRA accession
      */
     public static boolean isValid(String acc) {
-        if (!isSupported()) {
-            return false;
+        boolean looksLikeSRA = false;
+        File f = new File(acc);
+        if (f.isFile()) {
+            try {
+                byte[] buffer = new byte[8];
+                byte[] signature1 = "NCBI.sra".getBytes();
+                byte[] signature2 = "NCBInenc".getBytes();
+
+                InputStream is = new FileInputStream(f);
```

Comment:
We're java 8 now!  So this can be made into a try-with resources and the close can be done implicitly.